pkg/scheduler/apis/config/v1beta2/defaults.go

```diff
@@ -80,8 +78,9 @@ func SetDefaults_KubeSchedulerConfiguration(obj *v1beta2.KubeSchedulerConfigurat
 				hostPort := net.JoinHostPort(*obj.HealthzBindAddress, strconv.Itoa(config.DefaultInsecureSchedulerPort))
 				obj.HealthzBindAddress = &hostPort
 			} else {
-				// TODO: in v1beta2 we should let this error instead of stomping with a default value
-				obj.HealthzBindAddress = &defaultBindAddress
+				// This is a malformed partial address and there's nothing we can do
```

Comment:
Isn't there any guards in validation.go?
Also I don't think the comment is necessary.

Reply:
You're right, it is caught by validation. However we didn't test this scenario in validation_test.go, so I've added a test for this.

Reply:
awesome